robot/Cumulus/resources/GiftEntryPageObject.py

```diff
@@ -447,3 +447,114 @@ def verify_record_picklist_values(self,field,*args,**kwargs):
+
+    def fill_modal_form(self,**kwargs):
+        """Fill the gift entry modal form fields with specified values.
+        Key is field name and value is value to be entered for field """
+        for key,value in kwargs.items():
+            locator=npsp_lex_locators["gift_entry"]["modal_id"].format(key)
+            type=self.selenium.get_element_attribute(locator,"data-qa-locator")
+            field_locator=npsp_lex_locators["gift_entry"]["modal_field"].format(key,"input")
+            print(f"type is {type}")
+            if 'input autocomplete' in type :
+                self.salesforce._populate_field(locator,value)
+                qa_id="Select "+value
+                option=npsp_lex_locators["gift_entry"]["modal_id"].format(qa_id)
+                self.selenium.wait_until_page_contains_element(option)
+                try:
+                    self.selenium.scroll_element_into_view(option)
+                    self.selenium.click_element(option)
+                except ElementNotInteractableException:
+                    self.salesforce._jsclick(option)
+            elif type.startswith("autocomplete"):
+                self.salesforce._populate_field(field_locator,value)
+                option=npsp_lex_locators["gift_entry"]["modallookup-option"].format(value)
+                self.selenium.wait_until_page_contains_element(option)
+                try:
+                    self.selenium.scroll_element_into_view(option)
+                    self.selenium.click_element(option)
+                except ElementNotInteractableException:
+                    self.salesforce._jsclick(option)
+            elif 'combobox' in type :
+                self.selenium.wait_until_page_contains_element(field_locator)
+                self.selenium.click_element(field_locator)
+                popup=npsp_lex_locators["flexipage-popup"]
+                self.selenium.wait_until_page_contains_element(popup)
+                option=npsp_lex_locators["span_button"].format(value)
+                self.selenium.scroll_element_into_view(option)
+                self.selenium.click_element(option)
+            elif 'textarea' in type :
+                field_locator=npsp_lex_locators["gift_entry"]["modal_field"].format(key,"textarea")
+                self.selenium.scroll_element_into_view(field_locator)
+                self.salesforce._populate_field(field_locator,value)
+            elif 'datetime' in type :
+                locator=npsp_lex_locators["bge"]["datepicker_open"].format("Date")
+                self.selenium.click_element(field_locator)
+                self.selenium.wait_until_page_contains_element(locator)
+                self.selenium.input_text(field_locator,value,clear=True)
+                self.selenium.input_text(field_locator,value)
+                # option=npsp_lex_locators["span_button"].format(value)
+                # self.selenium.click_element(option)
```

Comment:
Can the commented out code be removed or any specific reason to keep ?

Reply:
Yeah it can be removed.